common/flatpak-context.c

```diff
@@ -488,6 +493,69 @@ flatpak_context_apply_generic_policy (FlatpakContext *context,
                        g_ptr_array_free (new, FALSE));
 }
 
+static gboolean
+validate_hex_uint16 (const char *value)
+{
+  long n;
+
+  g_assert (value != NULL);
+
+  if (strlen (value) > 4)
+    return FALSE;
+
+  n = strtol (value, NULL, 16);
```

Comment:
This allows trailing non-hex.

Reply:
Good point. I'll add a check for the end pointer position as well, it should cover this case

Reply:
Done